src/pset/mod.rs

```diff
@@ -248,13 +253,296 @@ impl PartiallySignedTransaction {
 
         Ok(())
     }
+
+    // Common pset blinding checks
+    fn blind_checks(
+        &self,
+        inp_txout_sec: &[Option<&TxOutSecrets>],
+    ) -> Result<
+        (
+            Vec<(u64, AssetBlindingFactor, ValueBlindingFactor)>,
+            Vec<usize>,
+        ),
+        PsetBlindError,
+    > {
+        if inp_txout_sec.len() != self.inputs.len() {
+            return Err(PsetBlindError::InputTxOutSecretLen);
+        }
+
+        let mut blind_out_indices = Vec::new();
```

Comment:
nit: can use `Vec::with_capacity(self.outputs.len())`

Reply:
It is bad to assign worst-case capacity when we might not end up using it? Suppose we have a tx with 4000 outputs and only 1 to blind, does this allocate memory for 4000 capacity?